tangram_bundler/__init__.py

```diff
@@ -1,100 +1,235 @@
-                if 'url' in weight:
-                    file_list.append(os.path.normpath(folder + '/' + weight['url']))
+    if 'fonts' in rootNode:
+        fontsNode = rootNode['fonts']
+        for font in fontsNode:
+            fontNode = fontsNode[font]
+            for prop in fontNode:
+                # initial Tangram font support
+                if 'url' in prop:
+                    fileList.append(os.path.relpath(prop['url'], basePath))
+                # newer Tangram font support
+                if 'weight' in prop and type(prop['weight']) is str:
+                    fileList.append(os.path.relpath(prop['weight'], basePath))
 
     # Search for textures urls
-    if 'textures' in yaml_file:
-        for texture in yaml_file['textures']:
-            if 'url' in yaml_file['textures'][texture]:
-                file_list.append(os.path.normpath(folder + '/' + yaml_file['textures'][texture]['url']))
-
-    # Search for u_envmap or u_ramp urls
-    if 'styles' in yaml_file:
-        for style in yaml_file['styles']:
-            if 'shaders' in yaml_file['styles'][style]:
-                if 'uniforms' in yaml_file['styles'][style]['shaders']:
-                    for uniform in yaml_file['styles'][style]['shaders']['uniforms']:
-                        addUniformTextureDependency(file_list, folder, yaml_file, style, uniform)
-
-    # Search for inner dependencies
-    if 'import' in yaml_file:
-        if (type(yaml_file['import']) is str):
-            addDependencies(file_list, folder + '/' + yaml_file['import'])
+    if 'textures' in rootNode:
+        texturesNode = rootNode['textures']
+        for texture in texturesNode:
+            textureNode = texturesNode[texture]
+            if 'url' in textureNode:
+                fileList.append(os.path.relpath(textureNode['url'], basePath))
+
+    # Search for asset url in styles
+    if 'styles' in rootNode:
+        stylesNode = rootNode['styles']
+        for style in stylesNode:
+            styleNode = stylesNode[style]
+            if 'shaders' in styleNode:
+                shadersNode = styleNode['shaders']
+                if 'uniforms' in shadersNode:
+                    uniformsNode = shadersNode['uniforms']
+                    for uniform in uniformsNode:
+                        addUniformTextureDependency(fileList, basePath, rootNode, style, uniform)
+            if 'texture' in styleNode:
+                fileList.append(os.path.relpath(styleNode['texture'], basePath))
+            if 'material' in styleNode:
+                materialNode = styleNode['material']
+                if (type(materialNode) is dict):
+                    for prop in ['emission', 'ambient', 'diffuse', 'specular', 'normal']:
+                        propNode = materialNode[prop]
+                        if (type(propNode) is dict and 'texture' in propNode):
+                            fileList.append(os.path.relpath(propNode['texture'], basePath))
+
```

Comment:
Since Tangram JS [v0.14.0](https://github.com/tangrams/tangram/releases/tag/v0.14.0), textures can also be specified inside `draw` groups for `points` and `lines`. Those should be accounted for here too.

Reply:
Ahh, we do not support this in ES, we should create an issue for this on ES. Will rectify, thanks.

Reply:
Interesting, since we do not have this logic in ES, I never thought through about this, but it seems like will have to go through all layers' all draw rules to check for `texture` draw rule. I know its too late in the discussion since JS has been supporting it for a while, but will it not make sense to restrict usage of these draw rule textures to "named" textures already defined in the scene's texture block?

Reply:
It looks like we did add support for this in Tangram ES https://github.com/tangrams/tangram-es/issues/1645

Reply:
Hmm I see, but in that case we are not doing it correctly in the importer, as its these texture urls are not considered for resolution with respect to the apt imported scene. Refer: https://github.com/tangrams/tangram-es/blob/e150471ade602e0f1e4f6d80e876937eb6ac05e5/core/src/scene/importer.cpp#L291

Reply:
There is not, will add one.